[DOCS] Add readiness listener documentation (#85464)
elastic/elasticsearch: docs/reference/setup/advanced-configuration.asciidoc

NOTE: If you are running {es} as a Windows service, you can change the heap size
using the service manager. See <<windows-service>>.

[[readiness-tcp-port]]
===== Enable the Elasticsearch TCP readiness port

NOTE: This feature is currently experimental for internal use by Elastic software only.

> If configured, a node can open a TCP port when the node is in a ready state. A node is deemed
> ready when it has successfully joined a cluster. In a single node configuration, the node is
> said to be ready, when it's able to accept requests.

Reviewer: ```suggestion
When the readiness TCP port is enabled, a client can connect to the port to determine whether or not the {es} node is in a ready state. 
A node is in a ready state once it has successfully joined a cluster. 
In a single node cluster, the node becomes ready once it can  accept requests.
```

> The readiness TCP port can be configured through the `readiness.port` setting and it is
> always bound to the loopback address. By default, the IPv4 loopback address of `127.0.0.1`
> is used. If you wish to bind the readiness port to the IPv6 loopback address of `::1`,
> you should add `-Djava.net.preferIPv6Addresses=true` to the <<set-jvm-options,JVM options>>.

Reviewer: ```suggestion
To enable the readiness TCP port, use the `readiness.port` setting. The port is
always bound to the loopback address, which defaults to the IPv4 loopback address`127.0.0.1`.
To bind the readiness port to the IPv6 loopback address `::1`,
add `-Djava.net.preferIPv6Addresses=true` to the <<set-jvm-options,JVM options>>.
```

If the node leaves the cluster, or the <<put-shutdown,Shutdown API>> is used to mark the node
for shutdown, the readiness port is immediately closed.

> When clients connect to the readiness TCP port, no data is sent back to the client, the server
> simply terminates the socket connection. A successful connection on the configured TCP port
> signals that the Elasticsearch node is ready, while not being able to connect means that the node
> is not ready.

Reviewer: ```suggestion
A successful connection to the readiness TCP port signals that the {es} node is ready. When a client connects to the readiness port, the server simply terminates the socket connection. No data is sent back to the client. 
If a client cannot connect to the readiness port, the node is not ready.

```
